sqlmock_test.go

```diff
@@ -354,6 +354,63 @@ func TestPreparedQueryExecutions(t *testing.T) {
 	}
 }
 
+func TestPreparedQueryMultipleExecutions(t *testing.T) {
+	t.Parallel()
+	db, mock, err := New()
+	if err != nil {
+		t.Errorf("an error '%s' was not expected when opening a stub database connection", err)
+	}
+	defer db.Close()
+
+	mock.MatchExpectationsInOrder(false)
+	mock.AllowRepeatedExpectationMatching(true)
+
+	rs := NewRows([]string{"id", "title"}).FromCSVString("5,hello world")
+	mock.ExpectPrepare("SELECT (.+) FROM articles WHERE id = ?").ExpectQuery().
+		WithArgs(5).
+		WillReturnRows(rs)
+
+	stmt, err := db.Prepare("SELECT id, title FROM articles WHERE id = ?")
+	if err != nil {
+		t.Errorf("error '%s' was not expected while creating a prepared statement", err)
+	}
+
+	stmt2, err := db.Prepare("SELECT id, title FROM articles WHERE id = ?")
+	if err != nil {
+		t.Errorf("error '%s' was not expected while creating a prepared statement", err)
+	}
+
+	var id1, id2 int
+	var title1, title2 string
+	err = stmt.QueryRow(5).Scan(&id1, &title1)
+	if err != nil {
+		t.Errorf("error '%s' was not expected querying row from statement and scanning", err)
+	}
+
+	err = stmt2.QueryRow(5).Scan(&id2, &title2)
+	if err != nil {
+		t.Errorf("error '%s' was not expected querying row from statement and scanning", err)
+	}
+
+	if id1 != 5 {
```

Comment:
if id1 != id2 || id1 != 5 same for title, not necessary to repeat same error twice